### ISSUE TITLE
boxsdk 2 not compatible with six==1.11.0

### ISSUE THREAD

**Issue by ivanpricewaycom** [box/box-python-sdk#239]
Hello,

a new version of the six module has been released, boxsdk is not compatible.

to replicate the problem:

pip install six==1.11.0 boxsdk==2.0.0a8
python -c "import boxsdk"

results in:
```
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/usr/local/lib/python2.7/site-packages/boxsdk/__init__.py", line 6, in <module>
    from .client import *  # pylint:disable=wildcard-import,redefined-builtin
  File "/usr/local/lib/python2.7/site-packages/boxsdk/client/__init__.py", line 5, in <module>
    from .client import Client
  File "/usr/local/lib/python2.7/site-packages/boxsdk/client/client.py", line 12, in <module>
    from ..object.events import Events
  File "/usr/local/lib/python2.7/site-packages/boxsdk/object/events.py", line 15, in <module>
    class EventsStreamType(with_metaclass(ExtendableEnumMeta, TextEnum)):
  File "/usr/local/lib/python2.7/site-packages/boxsdk/util/compat.py", line 77, in with_metaclass
    return type.__new__(TemporaryMetaSubclass, str('temporary_class'), bases, {})
TypeError: metaclass conflict: the metaclass of a derived class must be a (non-strict) subclass of the metaclasses of all its bases
```

I reported the bug in six and they have responded, suggesting a change in box:
https://github.com/benjaminp/six/issues/210#issuecomment-330206799

our workaround for the moment is to pin six to version 1.10, i link to this here for posterity.

cheers

**Comment by jmoldow** [comment]
Thanks for reporting this! I have fixes #240 and #241 which we'll try to merge and release sometime today.

**Comment by hcbomb** [comment]
@jmoldow Did a fresh install of pip, python27 and boxsdk and still getting this error. My workaround for now is just to downgrade back to six v1.10.0

**Comment by jmoldow** [comment]
@hcbomb I just released versions 1.5.5 and 2.0.0a9 to PyPI, which have the fix in them.
